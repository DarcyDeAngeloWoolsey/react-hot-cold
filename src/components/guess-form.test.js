import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

import GuessForm from './guess-form';

describe('<GuessForm />', () => {
    it('Renders without crashing', () => {
        shallow(<GuessForm />);
    });
        
//    it('Renders correctly', () => {
//        const tree = renderer.create(
//            <GuessForm />
//        ).toJSON();
//        expect(tree).toMatchSnapshot();
//    });
        
    it('Find the form and simulate a submit', () => {
        const callback = jest.fn();
        const wrapper = mount(<GuessForm    onGuess={callback} />
         );
        
        wrapper.instance().setEditing(true);
        wrapper.update();
        wrapper.find('input[name="userGuess"]').instance().value = '71';

      wrapper.find('form').simulate('submit')

        expect(callback).toHaveBeenCalledWith('71');
    })
        
});