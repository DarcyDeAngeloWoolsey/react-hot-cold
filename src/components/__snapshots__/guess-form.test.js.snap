exports[`<GuessForm /> Renders correctly 1`] = `
<form
  onSubmit={[Function]}>
  <label
    htmlFor="userGuess">
    Enter your Guess
  </label>
  <input
    autoComplete="off"
    className="text"
    id="userGuess"
    maxLength="3"
    name="userGuess"
    placeholder={32}
    required={true}
    type="text" />
  <input
    className="button"
    id="guessButton"
    name="submit"
    type="submit"
    value="Guess" />
</form>
`;
